src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -7161,6 +7161,98 @@ additionalReferences:=s_valueTupleRefs)
                        Sub(t) Assert.Equal(SpecialType.System_Int32, t.SpecialType))
         End Sub
 
+        <Fact>
+        <WorkItem(258853, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/258853")>
+        Public Sub BadOverloadWithTupleLiteralWithNaturalType()
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Module C
+    Sub M(x As Integer)
+    End Sub
+    Sub M(x As String)
+    End Sub
+
+    Sub Main()
+        M((1, 2))
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:=s_valueTupleRefs)
+
+
+            compilation.AssertTheseDiagnostics(
+<errors>
+    BC30518: Overload resolution failed because no accessible 'M' can be called with these arguments:
+    'Public Sub M(x As Integer)': Value of type '(Integer, Integer)' cannot be converted to 'Integer'.
+    'Public Sub M(x As String)': Value of type '(Integer, Integer)' cannot be converted to 'String'.
+        M((1, 2))
+        ~
+</errors>)
+        End Sub
+
+        <Fact>
+        <WorkItem(258853, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/258853")>
+        Public Sub BadOverloadWithTupleLiteralWithNothing()
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Module C
+    Sub M(x As Integer)
+    End Sub
+    Sub M(x As String)
+    End Sub
+
+    Sub Main()
+        M((1, Nothing))
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:=s_valueTupleRefs)
+
+
+            compilation.AssertTheseDiagnostics(
+<errors>
+    BC30518: Overload resolution failed because no accessible 'M' can be called with these arguments:
+    'Public Sub M(x As Integer)': Value of type '(Integer, Object)' cannot be converted to 'Integer'.
+    'Public Sub M(x As String)': Value of type '(Integer, Object)' cannot be converted to 'String'.
+        M((1, Nothing))
+        ~
+</errors>)
+        End Sub
+
+        <Fact>
+        <WorkItem(258853, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/258853")>
+        Public Sub BadOverloadWithTupleLiteralWithAddressOf()
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Module C
+    Sub M(x As Integer)
+    End Sub
+    Sub M(x As String)
+    End Sub
+
+    Sub Main()
+        M((1, AddressOf Main))
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:=s_valueTupleRefs)
+
+
+            compilation.AssertTheseDiagnostics(
+<errors>
+    BC30518: Overload resolution failed because no accessible 'M' can be called with these arguments:
+    'Public Sub M(x As Integer)': Expression does not produce a value.
```

Comment:
>Public Sub M(x As Integer)': Expression does not produce a value. [](start = 5, length = 65)

This shouldn't block this PR, but it feels like we should be getting a friendlier message here. Something about AddressOf.